### PR TITLE
update mask logic to keep up with current workflow

### PR DESCRIPTION
**Pull request by chiahaoliu**

### CODE REVIEW

**Comment by CJ-Wright** [comment]
Can you import the db builder from simulation into conftest if that is where it's new home is?
Also can you change the fixture scopes to 'module'? I think this will speed up the tests and give us a lower total overhead.

**Comment by chiahaoliu** [comment]
@CJ-Wright  please check if that works

File: xpdan/data_reduction.py [hunk 1]
```diff
@@ -152,10 +152,17 @@ def _prepare_header_list(headers):
     return header_list
 
 
-def _load_config():
-    with open(
-            os.path.join(an_glbl.config_base, an_glbl.calib_config_name)) as f:
-        config_dict = yaml.load(f)
+def _load_config(header):
```
Comment (CJ-Wright) [inline xpdan/data_reduction.py]: I think you also need to change line [247](https://github.com/xpdAcq/xpdAn/pull/32/files#diff-79db28d59f034ae0a4b4bc04efb5c88eR247) to take the header

**Comment by chiahaoliu** [comment]
@CJ-Wright  @sbillinge 

Is it possible to review this PR? It contains changes reflected current documentation and of course, we could discuss implementation later.

File: xpdan/data_reduction.py [hunk 2]
```diff
@@ -272,20 +289,28 @@ def integrate_and_save(headers, dark_sub_bool=True,
...
+                mask = None
+            mask_fn = os.path.splitext(f_name)[0]  # remove ext
+            if mask is not None:
                 print("INFO: mask file '{}' is saved at {}"
-                      .format(f_name, root_dir))
-                np.save(os.path.join(root_dir, f_name),
-                        mask)
+                      .format(mask_fn, root_dir))
+                np.save(os.path.join(root_dir, mask_fn),
+                        ~mask)  # default is .npy from np.save
```
Comment (CJ-Wright) [inline xpdan/data_reduction.py]: Why are you saving the `~mask` rather than the mask?

File: xpdan/data_reduction.py [hunk 3]
```diff
@@ -299,7 +324,10 @@ def integrate_and_save(headers, dark_sub_bool=True,
                                    [chi_fn_Q, chi_fn_2th],
                                    [header_rv_list_Q, header_rv_list_2theta]):
                 print("INFO: save chi file: {}".format(fn))
-                rv = ai.integrate1d(img, npt, filename=fn, mask=~mask,
+                if mask is not None:
+                    # flip before handing in mask
+                    mask = ~mask
```
Comment (CJ-Wright) [inline xpdan/data_reduction.py]: I would put the `~mask` logic in the integrate1d iteslf. This way the mask is consistent with the skbeam masking conventions to almost the end. Does this mean that if the mask is retrieved from the header it is already flipped?

Comment (chiahaoliu) [inline xpdan/data_reduction.py]: No, I only flip it when we need to save. As user want to visualize their mask with existing plotting program, which follows different convention.

File: xpdan/data_reduction.py (hunk 2 above)
Comment (chiahaoliu) [inline xpdan/data_reduction.py]: As user want to visualize their mask with existing plotting program, which follows different convention. Current mask is different from the convention with ``pyFAI`` and ``fit2D``. 
should we rethink about the convention?

@sbillinge  any comment?

File: xpdan/data_reduction.py (hunk 3 above)
Comment (CJ-Wright) [inline xpdan/data_reduction.py]: I would seriously advise against this, as whatever happens we will no longer be able to  load our masks into our own functions.

**Comment by chiahaoliu** [comment]
@CJ-Wright  hey, I remove my weird flipping logic. Could you review if you have time? Thanks.

**Comment by CJ-Wright** [comment]
Looks good to me!

@chiahaoliu I don't consider the flipped mask 'weird', there are pros and cons to each convention. I just think that the convention issue deserves its own issue/PR/discussion. I know that this PR had a lot of mission critical code so I didn't want to hold it up over a convention change.

**Pull request merged**